src/util/dstring.h

```diff
@@ -25,11 +27,11 @@ class dstringt
   }
 
   // this is safe for static objects
-  // the 2nd argument is to avoid accidental conversions
+  // marked explicit to avoid accidental conversions
   #ifdef __GNUC__
   constexpr
   #endif
-  dstringt(unsigned _no, unsigned):no(_no)
+  explicit dstringt(unsigned _no):no(_no)
   {
   }
 
```

Comment:
Note that the second parameter is deliberate, even over "explicit". This is meant to stop you from writing dstring(1), etc.